static/js/services/generate-search-query.js

```diff
@@ -1,140 +1,133 @@
-        if (selected &&
-            options &&
-            selected.length > 0 &&
-            selected.length < options.length) {
-          return selected;
-        }
-      };
-
-      var formatContactsType = function(filters) {
-        var selected = getSelected(filters.types);
+      }
+    };
+
+    var formatErrors = function(filters) {
+      if (filters.valid === true) {
+        return { errors: 0 };
+      }
+      if (filters.valid === false) {
         return {
-          type: selected || ContactSchema.getTypes()
+          $operator: 'not',
+          $operands: { errors: 0 }
         };
-      };
-
-      var formatForm = function(filters) {
-        var selected = getSelected(filters.forms);
-        if (selected) {
-          return {
-            form: _.pluck(selected, 'code')
-          };
-        }
-      };
-
-      var formatErrors = function(filters) {
-        if (filters.valid === true) {
-          return { errors: 0 };
-        }
-        if (filters.valid === false) {
-          return {
-            $operator: 'not',
-            $operands: { errors: 0 }
-          };
-        }
-      };
-
-      var formatVerified = function(filters) {
-        if (filters.verified === true) {
-          return { verified: true };
-        }
-        if (filters.verified === false) {
-          return { verified: false };
-        }
-      };
-
-      var formatClinics = function(filters) {
-        var selected = getSelected(filters.facilities);
-        if (selected) {
-          return { clinic: selected };
-        }
-      };
-
-      var formatFreetext = function(filters) {
-        var freetext = filters.search;
-        if (freetext && freetext.indexOf(':') === -1) {
-          freetext += '*';
-        }
-        return freetext;
-      };
-
-      var TYPES = {
-        reports: {
-          buildQuery: function(filters, operands) {
-            operands.push(formatFreetext(filters));
-            operands.push(formatReportedDate(filters));
-            operands.push(formatReportType());
-            operands.push(formatClinics(filters));
-            operands.push(formatForm(filters));
-            operands.push(formatErrors(filters));
-            operands.push(formatVerified(filters));
-          },
-          schema: {
-            errors: 'int',
-            verified: 'boolean',
-            reported_date: 'date'
-          }
+      }
+    };
+
+    var formatVerified = function(filters) {
+      if (filters.verified === true) {
+        return { verified: true };
+      }
+      if (filters.verified === false) {
+        return { verified: false };
+      }
+    };
+
+    var formatClinics = function(filters) {
+      var selected = getSelected(filters.facilities);
+      if (selected) {
+        return { clinic: selected };
+      }
+    };
+
+    var formatFreetext = function(filters) {
+      var freetext = filters.search;
+      if (freetext && freetext.indexOf(':') === -1) {
```

Comment:
In the spirit of boyscouting, I think this line deserves a comment or function extraction with a more descriptive name. I'm guessing this has something to do with how you can search for `foo` and also `name:foo`, but I don't completely get it.